### PR TITLE
Multiple updates

### PR DESCRIPTION
**Pull request by amarzavery**
- Updated the codegenerator to process extensible enums as seen [here](https://github.com/Azure/autorest.nodejs/commit/c698c41c91798a4fccc421ee17feb3d2364c6497#diff-99c896ba66d1a5f58410c2e2a5a02790R51)
- Added tests for extensible enums
- Taking a dependency on [2.3.44 version of the modeler](https://github.com/Azure/autorest.nodejs/compare/master...amarzavery:testfix?expand=1#diff-04c6e90faac2675aa89e2176d2eec7d8R20) that will provide additionalproperties Dictionary also when `"type": "object"` is not present on a model definition
- Added support to [de]serialize additionalProperties if specified in the swagger
- Added tests to verify additionalProperties get [de]serialized properly

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: src/vanilla/ClientModelExtensions.cs [hunk 1]
```diff
@@ -47,7 +48,7 @@ public static string GetFormattedReferenceValue(this Parameter parameter)
 
             PrimaryType primaryType = sequence.ElementType as PrimaryType;
             EnumType enumType = sequence.ElementType as EnumType;
-            if (enumType != null && enumType.ModelAsString)
+            if (enumType != null && (enumType.ModelAsString || enumType.ModelAsExtensible))
```
Comment (amarzavery) [inline src/vanilla/ClientModelExtensions.cs]: @fearthecowboy - Added check for extensible enums. Is this correct?

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.17*

## ✔️ General: 100%



## ✔️ Azure: 100%

**Comment by amarzavery** [comment]
@fearthecowboy - coverage for node.js codegenerator vanilla and azure @ 💯%

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.3.17*

## ✔️ General: 100%



## ✔️ Azure: 100%

**Pull request merged**